### ISSUE TITLE
Inconsistent status code

### ISSUE THREAD

**Issue by diegoatpitch** [imgproxy/imgproxy#1260]
This issue came to my attention when an external agent started poking at our instance. We saw an elevated number of responses with 500 status code from Imgproxy. Upon investigation we see that some of the URLs the third party was trying would cause a 500. Trying some of the patterns I saw that would lead to a 500 I came to this scenario:

Imgproxy thinks the source URL is base64 encoded and ends up trying to use it. Some combinations of URLs returns 404, other 500. Message for 404 varies too.

- `/_/a/YnVja2V0L29iamVjdA` → 500 "Internal error"
- `/_/b/YnVja2V0L29iamVjdA` → 500 "Internal error" (Any other one letter segment before the base64 part results in 500, expect `/f/`. See below)
- `/_/f/YnVja2V0L29iamVjdA` → 404 "Source image is unreachable"
- `/_/aa/YnVja2V0L29iamVjdA` → 404 "Source image is unreachable" (Other non-one-letter values result in 404)

I would expect the status code to be consistently 404 in all cases. Not sure this is a bug or some behavior I am not seeing.

Some other values that fails with 500:

- `/.circleci/ssh-config`
- `/.git/config`
- `/gateway/routes`
- `/api/v1/pods`

Is this something Imgproxy can handle better or should I put other measure in place to contain the 500s? Thanks.

**Comment by DarthSim** [comment]
Hey @diegoatpitch!

I can't reproduce this with the `latest` build. Which version do you use? And what errors do you see in the logs?

**Comment by diegoatpitch** [comment]
I am using version 3.23.0, which I think is `latest`.

My bad, let me add more context to this.

This is the config I am using, at least the ones that impact the Source URL.

```env
IMGPROXY_ALLOWED_SOURCES=s3://
IMGPROXY_BASE_URL=s3://
```

This is one of the logs I see with 500 in DataDog:

```json
{
	"id": "...",
	"content": {
		"timestamp": "...",
		"tags": [],
		"service": "imgproxy",
		"message": "Completed in 312.048µs /editor_tools/module",
		"attributes": {
			"stack": "/app/processing_handler.go:350 main.handleProcessing.func2\n/app/processing_handler.go:336 main.handleProcessing\n/app/server.go:138 main.buildRouter.withCORS.func4\n/app/server.go:114 main.buildRouter.withDevRequestLimit.func5\n/app/server.go:215 main.withPanicHandler.func1\n/app/server.go:127 main.buildRouter.withMetrics.func6\n/app/router/router.go:156 github.com/imgproxy/imgproxy/v3/router.(*Router).ServeHTTP\n/usr/local/go/src/net/http/server.go:3138 net/http.serverHandler.ServeHTTP\n/usr/local/go/src/net/http/server.go:2040 net/http.(*conn).serve\n/usr/local/go/src/runtime/asm_arm64.s:1223 runtime.goexit",
			"method": "GET",
			"level": "error",
			"service": "imgproxy",
			"client_ip": "157.245.89.119",
			"source": "stdout",
			"time": "2024-03-22T09:25:20Z",
			"error": "Can't download source image: Get \"s3://%9A%87n%95\": operation error S3: GetObject, serialization failed: serialization failed: input member Key must not be empty",
			"request_id": "YZLXzglgHMe8ahNk23YHX",
			"timestamp": 1711099520212,
			"status": 500
		}
	}
}
```

**Comment by DarthSim** [comment]
Ah, it makes sense now, thanks. Added checks for bucket name and object key to the `latest` build.

**Comment by DarthSim** [comment]
The fix is released in `v3.24.0`

**Comment by diegoatpitch** [comment]
Thanks. Confirmed that it solved the issue.
